widgetastic_manageiq.py

```diff
@@ -878,3 +875,57 @@ def last_page(self):
     @property
     def items_amount(self):
         return self.paginator.page_info()[1]
+
+
+class AlertEmail(Widget):
+
+    ROOT = "//div[@id='edit_email_div']"
+    ADD_BUTTON = ".//div[@title='Add']"
+    RECIPIENTS = "./div[@id='edit_to_email_div']//a"
+    EMAIL_LINK = ".//a[text()='{}']"
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.RECIPIENTS_INPUT = TextInput(self, "email")
+
+    def __locator__(self):
+        return self.ROOT
+
+    def fill(self, values):
+        if isinstance(values, basestring):
+            values = [values]
+        elif isinstance(values, list):
+            pass
+        else:
+            raise ValueError("Emails should be a string or a list")
+        if self.read() == set(values):
+            return False
+        else:
+            values_to_remove = self._values_to_remove(values)
+            values_to_add = self._values_to_add(values)
+            for value in values_to_remove:
+                self._remove_recipient(value)
+            for value in values_to_add:
+                self._add_recipient(value)
+            return True
+
+    @property
+    def button_add_recipient(self):
+        return self.browser.element(self.ADD_BUTTON, parent=self)
+
+    def _values_to_remove(self, values):
+        return list((set(values) ^ self.read()) - set(values))
+
+    def _values_to_add(self, values):
+        return list((set(values) ^ self.read()) - self.read())
+
+    def _add_recipient(self, email):
+        self.RECIPIENTS_INPUT.fill(email)
+        self.button_add_recipient.click()
```

Comment:
This does not invoke Widgetastic's click. You need to get rid of button_add_recipient property and call the click like this:

```python
self.browser.click(self.ADD_BUTTON)
```

But it would be better to convert this to a View that contains widgets, as I mentioned before...